src/ray/raylet/scheduling/cluster_resource_manager.h

```diff
@@ -129,7 +129,12 @@ class ClusterResourceManager {
     return nodes_.count(node_id) > 0;
   }
 
-  void DebugString(std::stringstream &buffer) const;
+  bool IsNodeDraining(const scheduling::NodeID &node_id) const {
+    const auto &node = map_find_or_die(nodes_, node_id);
```

Comment:
is this safe? how do we know node id is always avialable from nodes_?

Reply:
Yes. If you check the callers, it's guaranteed that the node id is available.